js/README.md

```diff
@@ -0,0 +1,83 @@
+
+Fibre-JS currently supports the following transport providers:
+
+ - WebUSB ([Chrome and Edge only](https://caniuse.com/webusb))
+
+## How to use
+
+So you want your website or Electron App to connect to some resource(s) using Fibre. Let's start with a simple HTML scaffolding.
+
+```HTML
+<!doctype html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <title>Fibre-JS Test</title>
+    </head>
+    <body>
+        <h1 id="statusText">not connected</h1>
+        <button id="connectBtn" disabled>Connect...</button>
+    </body>
+</html>
+```
+
+Now, right before the closing `</body>`, insert a two script tags to load and open Fibre-JS.
+
+```HTML
+        <script src="fibre.js"></script>
+        <script>
+            fibreOpen().then((libfibre) => {
+                // Fibre-JS is ready to use!
+            });
+        </script>
+```
+
+This will load the files `fibre.js`, `libfibre-wasm.js` and `libfibre-wasm.wasm` from the same directory the HTML file comes from so make sure those files exist.
+
+Now that Fibre-JS is ready to use we can start discovering remote objects. For this we need to specify a filter string which tells Fibre where to look for objects. In this example we want to talk to a USB device that runs Fibre (more specfically an [ODrive](https://odriverobotics.com/)). The parameters we use here are something you can find out from reading the docs of your USB device or inspecting the device by using `lsusb` on a Linux or macOS computer.
+
+```JS
+const filter = 'usb:idVendor=0x1209,idProduct=0x0D32,bInterfaceClass=0,bInterfaceSubClass=1,bInterfaceProtocol=0';
+let onFoundObject = async (obj) => {
+    console.log("found an object!", obj);
+    // TODO: do something with obj
+}
+libfibre.startDiscovery(filter, onFoundObject);
+```
+
+Now that we have the object we can start calling functions on it. Let's assume our object has the property `vbus_voltage`. In the `onFoundObject` handle, add:
+
+```JS
+while (true) {
+    document.getElementById("statusText").innerHTML = "vbus_voltage: " + (await obj.vbus_voltage.read());
+    await new Promise((resolve) => setTimeout(resolve, 100));
+}
+```
+
+This will read the `vbus_voltage` property every 100ms and display the value on the web page.
+
+Optionally you could add a callback to the `obj._onLost` promise in order to gracefully stop polling when the device disconnects.
+
+Since we're using WebUSB in this example, there's one more thing left to do: The user needs to authorize the website (once only) to access the USB device. This must happen on a user gesture, which is why earlier we put a Connect-button into our HTML.
+
+So directly after the call to `libfibre.startDiscovery()`, add:
+
+```JS
+document.getElementById("connectBtn").onclick = libfibre.usbDiscoverer.showDialog;
+document.getElementById("connectBtn").removeAttribute('disabled');
```

Comment:
Did you mean to set the attribute here, instead of remove?

Reply:
The `removeAttribute` is intentional in order to make button go from being disabled to not being disabled. Maybe this is an unusual way of achieving this?